### DISCUSSION TITLE
SCIP solver get error result for MIP

### DISCUSSION THREAD

**Discussion by yu961549745** [google/or-tools#2295] [Linear Solver questions] [answered]
**What version of OR-tools and what language are you using?**
Version: 8.1.8487
Language: Python

**Which solver are you using** 
SCIP

**What operating system (Linux, Windows, ...) and version?** 
MacOs

**What did you do?**

```python
from ortools.linear_solver import pywraplp

solver = pywraplp.Solver.CreateSolver('SCIP')

x0000 = solver.IntVar(2, 8, 'x0000')
x0001 = solver.IntVar(0, 8, 'x0001')
x0002 = solver.IntVar(0, 8, 'x0002')
x0003 = solver.BoolVar('x0003')
x0004 = solver.BoolVar('x0004')
x0005 = solver.BoolVar('x0005')

solver.Maximize(+ 1.0 * x0000 + 1.0 * x0001 + 1.0 * x0002 - 0.5 * x0004 - 0.5 * x0005)

solver.Add(+ 1.0 * x0000 + 1.0 * x0001 + 1.0 * x0002 <= 16.0)
solver.Add(+ 1.0 * x0000 - 8.0 * x0003 <= 0.0)
solver.Add(+ 1.0 * x0001 - 8.0 * x0004 <= 0.0)
solver.Add(+ 1.0 * x0002 - 8.0 * x0005 <= 0.0)
solver.Add(+ 1.0 * x0000 - 1.0 * x0003 >= 0.0)
solver.Add(+ 1.0 * x0001 - 1.0 * x0004 >= 0.0)
solver.Add(+ 1.0 * x0002 - 1.0 * x0005 >= 0.0)
solver.Add(+ 1.0 * x0003 + 1.0 * x0004 + 1.0 * x0005 >= 2.0)
solver.Add(- 0.5 * x0000 + 0.5 * x0001 + 0.5 * x0002 >= -0.0)
solver.Add(+ 0.5 * x0000 - 0.5 * x0001 + 0.5 * x0002 >= -0.0)
solver.Add(+ 0.5 * x0000 + 0.5 * x0001 - 0.5 * x0002 >= -0.0)

status = solver.Solve()
print(solver.ExportModelAsLpFormat(False))
assert status == pywraplp.Solver.OPTIMAL
assert 15.5 == solver.Objective().Value()
```

the optimal solution should be 15.5, but return 15.0.
change solver to `CBC` will get correct solution.

using java has the same issuse.

**Comment by ghost** [comment, marked as answer]
Gonna copy my answer from here: https://stackoverflow.com/questions/65244692/non-optimal-result-from-mip-program-in-google-or-tools/65244914#65244914

> It works well for CBC and SAT, looks like you have to set the PRIMAL_TOLERANCE for SCIP.
```python
solver_parameters = pywraplp.MPSolverParameters()
solver_parameters.SetDoubleParam(pywraplp.MPSolverParameters.PRIMAL_TOLERANCE, 0.001)
status = solver.Solve(solver_parameters)
```

**Comment by yu961549745** [comment]
> Gonna copy my answer from here: https://stackoverflow.com/questions/65244692/non-optimal-result-from-mip-program-in-google-or-tools/65244914#65244914
> 
> > It works well for CBC and SAT, looks like you have to set the PRIMAL_TOLERANCE for SCIP.
> 
> ```python
> solver_parameters = pywraplp.MPSolverParameters()
> solver_parameters.SetDoubleParam(pywraplp.MPSolverParameters.PRIMAL_TOLERANCE, 0.001)
> status = solver.Solve(solver_parameters)
> ```

works well for me.

**Comment by ghost** [comment]
Maybe lowering the default primal_tolerance should be a feature request?

**Reply by yu961549745** [reply]
I think so
